File: math/euclids_gcd/Nim/euclids_gcd.nim
import strutils

proc gcd(x, y: int): int =
    if x == 0:
      return y
    gcd(y mod x, x)

when isMainModule:
    echo "Enter first number"
    let num1 = readLine(stdin).parseInt()
    echo "Enter second number"    
    let num2 = readLine(stdin).parseInt()
    echo "GCD: ", gcd(num1, num2)

File: math/euclids_gcd/Rust/euclids_gcd.rs
fn gcd(mut a: i32, mut b: i32) -> i32 {
    while b != 0 {
        let temp = a % b;
        a = b;
        b =  temp;       
    }
    a
}

fn main() {
    let a = 10;
    let b = 15;
    println!("GCD({}, {}) = {}",a,b,gcd(a,b));
    
    let a = 35;
    let b = 10;
    println!("GCD({}, {}) = {}",a,b,gcd(a,b));

    
    let a = 31;
    let b = 2;
    println!("GCD({}, {}) = {}",a,b,gcd(a,b));
}
